src/compose/ComposeMenu.js

```diff
@@ -58,13 +57,18 @@ export const chooseUploadImageFilename = (uri: string, fileName: ?string): strin
   return name;
 };
 
-class ComposeMenu extends PureComponent<Props> {
-  uploadFile = (uri: string, fileName: ?string) => {
-    const { dispatch, destinationNarrow } = this.props;
-    dispatch(uploadFile(destinationNarrow, uri, chooseUploadImageFilename(uri, fileName)));
-  };
+export default function ComposeMenu(props: Props) {
+  const { destinationNarrow } = props;
+  const dispatch = useDispatch();
+
+  const uploadFileCallback = useCallback(
```

Comment:
nit: This `useCallback` isn't doing much; `uploadFileCallback`s' callers are functions that get recreated each time the `ComposeMenu` renders (they aren't using `useCallback`), and those functions are being passed to various children.